MeshGeoToolsLib/BoundaryElementsAlongPolyline.cpp

```diff
@@ -88,18 +89,38 @@ bool BoundaryElementsAlongPolyline::includesAllEdgeNodeIDs(const std::vector<std
     return (j==edge.getNumberOfBaseNodes());
 }
 
-MeshLib::Element* BoundaryElementsAlongPolyline::modifyEdgeNodeOrdering(const MeshLib::Element &edge, const GeoLib::Polyline &ply, const std::vector<std::size_t> &edge_node_distances_along_ply, const std::vector<std::size_t> &node_ids_on_poly) const
+MeshLib::Element* BoundaryElementsAlongPolyline::modifyEdgeNodeOrdering(
+    const MeshLib::Element& edge, const GeoLib::Polyline& ply,
+    const std::vector<std::size_t>& edge_node_distances_along_ply,
+    const std::vector<std::size_t>& node_ids_on_poly) const
 {
-    MeshLib::Element* new_edge = const_cast<MeshLib::Element*>(&edge);
-    // The first node of the edge should be always closer to the beginning of the polyline than other nodes.
-    // Otherwise, create a new element with reversed local node index
-    if (edge_node_distances_along_ply.front() > edge_node_distances_along_ply.back()
-            || (ply.isClosed() && edge_node_distances_along_ply.back() == node_ids_on_poly.size()-1)) {
-        MeshLib::Node** new_nodes = new MeshLib::Node*[edge.getNumberOfBaseNodes()];
-        std::reverse_copy(edge.getNodes(), edge.getNodes()+edge.getNumberOfBaseNodes(), new_nodes);
-        new_edge = new MeshLib::Line(new_nodes);
+    // The first node of the edge should be always closer to the beginning of
+    // the polyline than other nodes.
+    if (edge_node_distances_along_ply.front() >
+            edge_node_distances_along_ply.back() ||
+        (ply.isClosed() &&
+         edge_node_distances_along_ply.back() == node_ids_on_poly.size() - 1))
+    {  // Create a new element with reversed local node index
+        auto new_nodes = new MeshLib::Node*[edge.getNumberOfNodes()];
+        if (auto const* e = dynamic_cast<MeshLib::Line const*>(&edge))
```

Comment:
Can be done, although I find this to be more explicit.
If some one also votes for a change, I will change it.